src/components/__tests__/DynamicWidgets.js

```diff
@@ -0,0 +1,420 @@
+import { createLocalVue, mount } from '@vue/test-utils';
+import DynamicWidgets from '../DynamicWidgets';
+import { __setState } from '../../mixins/widget';
+import { plugin } from '../../plugin';
+jest.mock('../../mixins/widget');
+
+const MockRefinementList = {
+  props: { attribute: { type: String } },
+  render(h) {
+    return h('div', {
+      attrs: {
+        'widget-name': 'ais-refinement-list',
+        attribute: this.attribute,
+      },
+    });
+  },
+};
+
+const MockMenu = {
+  props: { attribute: { type: String } },
+  render(h) {
+    return h('div', {
+      attrs: { 'widget-name': 'ais-menu', attribute: this.attribute },
+    });
+  },
+};
+
+const MockHierarchicalMenu = {
+  props: { attributes: { type: Array } },
+  render(h) {
+    return h('div', {
+      attrs: {
+        'widget-name': 'ais-hierarchical-menu',
+        attributes: this.attributes,
+      },
+    });
+  },
+};
+
+it('renders all children without state', () => {
+  const localVue = createLocalVue();
+
+  localVue.use(plugin);
+
+  __setState(null);
+
+  const wrapper = mount(DynamicWidgets, {
+    localVue,
+    propsData: {
+      transformItems: items => items,
+    },
+    slots: {
+      default: `
+      <ais-refinement-list attribute="test1"/>
+      <ais-menu attribute="test2"/>
+      <ais-panel>
+        <ais-hierarchical-menu :attributes="['test3', 'test4']" />
+      </ais-panel>
+      `,
+    },
+    stubs: {
+      'ais-refinement-list': MockRefinementList,
+      'ais-menu': MockMenu,
+      'ais-hierarchical-menu': MockHierarchicalMenu,
+    },
+  });
+
+  expect(wrapper.html()).toMatchInlineSnapshot(`
+
+<div hidden="hidden"
+     class="ais-DynamicWidgets"
+>
+  <div class="ais-DynamicWidgets-widget">
+    <div widget-name="ais-refinement-list"
+         attribute="test1"
+    >
+    </div>
+  </div>
+  <div class="ais-DynamicWidgets-widget">
+    <div widget-name="ais-menu"
+         attribute="test2"
+    >
+    </div>
+  </div>
+  <div class="ais-DynamicWidgets-widget">
+    <div class="ais-Panel">
+      <div class="ais-Panel-body">
+        <div widget-name="ais-hierarchical-menu"
+             attributes="test3,test4"
+        >
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+`);
+});
+
+it('renders nothing without children', () => {
+  __setState({
+    attributesToRender: [],
+  });
+
+  const wrapper = mount(DynamicWidgets, {
+    propsData: {
+      transformItems: items => items,
+    },
+  });
```

Comment:
If I understand correctly, this test is about not providing any children `slots: { default: ... }` and seeing if it renders an empty container. So setting attributesToRender to an empty array isn't relevant. With this, we don't know the reason why the empty container is there is because of attributesToRender: [] or no children.

Reply:
solved in c0d1823